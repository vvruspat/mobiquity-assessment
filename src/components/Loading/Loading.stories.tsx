import { ComponentStory, ComponentMeta } from "@storybook/react";

import "../../index.css";
import { Loading } from "./Loading";

export default {
  title: "Components/Loading",
  component: Loading,
} as ComponentMeta<typeof Loading>;

const Template: ComponentStory<typeof Loading> = (args) => (
  <Loading {...args} />
);

export const LoadingTemplate = Template.bind({});

LoadingTemplate.args = {};
