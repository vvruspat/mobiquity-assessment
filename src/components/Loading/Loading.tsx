import {
  DetailedHTMLProps,
  FC,
  HTMLAttributes,
  PropsWithChildren,
} from "react";
import cn from "clsx";

import { ReactComponent as LoadingIcon } from "./assets/loading.svg";
import s from "./Loading.module.css";

type LoadingProps = DetailedHTMLProps<
  HTMLAttributes<HTMLDivElement>,
  HTMLDivElement
>;

/**
 * Loading component
 */
export const Loading: FC<PropsWithChildren<LoadingProps>> = ({
  className,
  ...htmlProps
}) => {
  return (
    <div className={cn(s.loading, className)} {...htmlProps}>
      <div className={s.loadingIcon}>
        <LoadingIcon />
      </div>
    </div>
  );
};
