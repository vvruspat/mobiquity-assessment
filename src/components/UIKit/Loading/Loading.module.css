.loading {
    padding: var(--padding-large);
}

.loadingIcon {
    display: flex;
    justify-content: center;
    align-items: center;
    color: var(--regular-color);
}