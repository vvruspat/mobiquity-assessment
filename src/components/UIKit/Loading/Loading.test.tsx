import { render, screen } from "@testing-library/react";
import { Loading } from "./Loading";

test("renders loading", () => {
  render(<Loading />);

  const el = screen.getByTestId("loading");

  expect(el).toMatchSnapshot();
});
