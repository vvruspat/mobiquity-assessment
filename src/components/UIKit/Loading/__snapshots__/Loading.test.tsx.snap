// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading 1`] = `
<div
  class="loading"
  data-testid="loading"
>
  <div
    class="loadingIcon"
  >
    <svg>
      loading.svg
    </svg>
  </div>
</div>
`;
